greenmail-core/src/main/java/com/icegreen/greenmail/smtp/commands/RcptCommand.java

```diff
@@ -27,7 +27,7 @@
  */
 public class RcptCommand
         extends SmtpCommand {
-    static Pattern param = Pattern.compile("RCPT TO:\\s?<([^>]+)>",
+    static Pattern param = Pattern.compile("RCPT TO:\\s?<([^>]+[^\\s]+)>",

```

Comment:
As I said, JavaMail ends up doing some bytes cutoff to force an old RFC and when it gets to this class, unencoded, the regex validation fails.

Anyway, I'm fine with reverting this change and force the user to encode the email address as I'm doing it on the test.